backend/src/controllers/entry/entry.ts

```diff
@@ -30,66 +45,40 @@ export const getAllEntries = async (req: Request, res: Response) => {
  * Request params journalId: string
  * Request body matches joiEntrySchema, but might be missing fields
  */
-export const createEntry = async (req: Request, res: Response, next: NextFunction) => {
+export const createEntry = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
   const { journalId } = req.params;
+  const entryData: UpdateEntryRequestBody = req.body;
 
-  // Ensure that the journal exists
-  const journal = await Journal.findById(journalId);
-  if (!journal) {
-    return next(new ExpressError('Journal not found.', 404));
-  }
-
-  validateEntryAnalysis(req, res, async (err: ExpressError) => {
-    if (err) {
-      return next(err); // Handle any validation errors
-    }
-
-    // If validation is successful, proceed to create the entry and analysis
-    const entryData = req.body;
-
-    const newEntry = new Entry({ journal: journalId, ...entryData });
-    const newAnalysis = new EntryAnalysis({
-      entry: newEntry.id,
-      analysis_content: entryData.analysis_content,
-    });
+  try {
+    const configId = await verifyJournalExists(journalId);
```

Comment:
Sure I like this idea!

Reply:
After trying it out, I think this actually makes things a little more complicated because the calling function now needs to do a null check on the returned `configId`. It seems a little simpler to let the controller functions catch all the errors and figure out what to do

Reply:
Let's go with your solution to un-nest the try-catch blocks. I think the actual solution here is to use middleware to verify the journal exists. So eventually we'd want to pull our this verification logic as described here in #203. This can be done after the TS migration though. Un-nesting the try-catch block now will set us up more nicely to get this done later.